src/renderer/rendered_window.rs

```diff
@@ -450,56 +443,59 @@ impl RenderedWindow {
+                self.scroll_delta = scroll_delta.round() as isize;
+            }
+            _ => {}
+        };
+    }
 
-                for i in 0..self.actual_lines.len() {
-                    let scrollback_index = (self.scrollback_top_index + i as isize)
-                        .rem_euclid(self.scrollback_lines.len() as isize)
-                        as usize;
-                    let actual_index = (self.actual_top_index + i as isize)
-                        .rem_euclid(self.actual_lines.len() as isize)
-                        as usize;
-                    self.scrollback_lines[scrollback_index] =
-                        self.actual_lines[actual_index].clone();
-                }
+    pub fn flush(&mut self, renderer_settings: &RendererSettings) {
+        let scroll_delta = self.scroll_delta;
+        self.scrollback_top_index += scroll_delta;
 
-                let mut scroll_offset = self.scroll_animation.position;
-
-                let minmax = self.scrollback_lines.len() - self.grid_size.height as usize;
-                log::trace!("Scroll offset {scroll_offset}, delta {scroll_delta}, minmax {minmax}");
-                // Do a limited scroll with empty lines when scrolling far
-                if scroll_delta.unsigned_abs() > minmax {
-                    let far_lines = renderer_settings
-                        .scroll_animation_far_scroll_lines
-                        .min(self.actual_lines.len() as u32)
-                        as isize;
-
-                    scroll_offset = (far_lines * scroll_delta.signum()) as f32;
-                    let empty_lines = if scroll_delta > 0 {
-                        self.actual_lines.len() as isize
-                            ..self.actual_lines.len() as isize + far_lines
-                    } else {
-                        -far_lines..0
-                    };
-                    for i in empty_lines {
-                        let i = (self.scrollback_top_index + i)
-                            .rem_euclid(self.scrollback_lines.len() as isize)
-                            as usize;
-                        self.scrollback_lines[i] = None;
-                    }
-                // And even when scrolling in steps, we can't let it drift too far, since the
-                // buffer size is limited
+        for i in 0..self.actual_lines.len() {
+            let scrollback_index = (self.scrollback_top_index + i as isize)
+                .rem_euclid(self.scrollback_lines.len() as isize)
+                as usize;
+            let actual_index = (self.actual_top_index + i as isize)
+                .rem_euclid(self.actual_lines.len() as isize)
+                as usize;
+            self.scrollback_lines[scrollback_index] = self.actual_lines[actual_index].clone();
+        }
+
+        if scroll_delta != 0 {
+            let mut scroll_offset = self.scroll_animation.position;
+
+            let minmax = self.scrollback_lines.len() - self.grid_size.height as usize;
```

Comment:
This variable name doesnt make sense to me. What is Minmax?

Reply:
I changed it to `max_delta` the absolute value of the allowed delta. So, `minmax` was short for "minimum or maximum"